openapi-cli/src/main/java/io/ballerina/openapi/generators/client/FunctionBodyGenerator.java

```diff
@@ -464,25 +465,40 @@ private  void createRequestBodyStatements(boolean isHeader, List<StatementNode>
     private void genStatementsForRequestMediaType(List<StatementNode> statementsList,
                                                   Map.Entry<String, MediaType> mediaTypeEntry) {
 
+        Schema requestBodySchema = mediaTypeEntry.getValue().getSchema();
         if (mediaTypeEntry.getKey().contains("json")) {
-            VariableDeclarationNode jsonVariable = generatorUtils.getSimpleStatement("json",
-                    "jsonBody", "check payload.cloneWithType(json)");
-            statementsList.add(jsonVariable);
-            ExpressionStatementNode expressionStatementNode = generatorUtils.getSimpleExpressionStatementNode(
-                    "request.setPayload(jsonBody)");
-            statementsList.add(expressionStatementNode);
+            if (requestBodySchema.get$ref() != null || requestBodySchema.getType() != null
+                    || requestBodySchema.getProperties() != null) {
+                VariableDeclarationNode jsonVariable = generatorUtils.getSimpleStatement("json",
```

Comment:
Is this condition OK? Whenever we have schema ref or we have a specific type or any property we go with below. 
```
if (requestBodySchema.get$ref() != null || requestBodySchema.getType() != null
                    || requestBodySchema.getProperties() != null) {
```
Can there be scenarios requestBodySchema.get$ref()== null and requestBodySchema.getProperties() != null? What do we get as requestBodySchema.getProperties()?

Reply:
I have used this if condition only to check whether the schema is empty or not. Since there is no direct property to know whether the schema is empty or not, we have to assume that request body schema has to be empty when all of the above conditions are false. 

Type of the request body payload does not matter here since the variable name is always `payload` for any type and in every scenario where type of the payload is not JSON we have to convert it to JSON to set in the request payload. 

```ballerina
json jsonBody = check payload.cloneWithType(json);
request.setPayload(jsonBody);
```